File: StudentGradingSystem.Api/Models/Subject.cs
using System.ComponentModel.DataAnnotations;

namespace StudentGradingSystem.Api.Models;

public class Subject
{
    public int Id { get; set; }

    [Required]
    [MaxLength(20)]
    public string SubjectCode { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [Range(0.5, 10)]
    public decimal Credits { get; set; }

    [Range(1, 8)]
    public int Semester { get; set; }

    public bool IsDeleted { get; set; } = false;

    public DateTime? DeletedAt { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public string? CreatedBy { get; set; }

    public string? UpdatedBy { get; set; }
}

File: student-grading-system-web/src/types/subject.ts
export interface Subject {
  id: number;
  subjectCode: string;
  name: string;
  credits: number;
  semester: number;
  departmentId: number;
  departmentName: string;
  facultyId: number;
  facultyName: string;
}

export interface CreateSubjectRequest {
  subjectCode: string;
  name: string;
  credits: number;
  semester: number;
  departmentId: number;
  facultyId: number;
}

export interface UpdateSubjectRequest {
  subjectCode: string;
  name: string;
  credits: number;
  semester: number;
  departmentId: number;
  facultyId: number;
}